admin/model/extension/payment/razorpay.php

```diff
@@ -0,0 +1,313 @@
+
+		if (!empty($data['filter_subscription_id'])) {
+			$implode = array();
+
+			$sub_statuses = explode(',', $data['filter_subscription_id']);
+
+			foreach ($sub_statuses as $sub_id) {
+				$implode[] = "subscription_id = '" . $sub_id . "'";
+			}
+
+			if ($implode) {
+				$sql .= " WHERE (" . implode(" OR ", $implode) . ")";
+			}
+		} elseif (isset($data['filter_plan_name']) && $data['filter_plan_name'] !== '') {
+			$sql .= " WHERE plan_id = '" . $data['filter_plan_name'] . "'";
+		} else {
+			$sql .= " WHERE plan_id > '0'";
+		}
+
+		// if (!empty($data['filter_date_created'])) {
+		// 	$sql .= " AND s.created_at = '" . (int)$data['filter_date_created'] . "'";
+		// }
+
+		// if (!empty($data['filter_customer'])) {
+		// 	$sql .= " AND CONCAT(firstname, ' ', lastname) LIKE '%" . $this->db->escape($data['filter_customer']) . "%'";
+		// }
+
+		if (!empty($data['filter_date_created'])) {
+			$sql .= " AND DATE(s.created_at) = DATE('" . $this->db->escape($data['filter_date_created']) . "')";
+		}
+
+		// if (!empty($data['filter_date_modified'])) {
+		// 	$sql .= " AND DATE(date_modified) = DATE('" . $this->db->escape($data['filter_date_modified']) . "')";
+		// }
+
+		// if (!empty($data['filter_total'])) {
+		// 	$sql .= " AND total = '" . (float)$data['filter_total'] . "'";
+		// }
+
+		$query = $this->db->query($sql);
+
+		return $query->row['total'];
+	}
+    public function getSubscriptionInfo($entity_id) {
+		//$query = $this->db->query("SELECT * FROM `" . DB_PREFIX . "razorpay_subscriptions` WHERE `entity_id` = '" . (int)$entity_id . "'");
+        $sql = "SELECT s.*,s.entity_id as sub_id,s.status as sub_status,s.created_at as sub_created,p.*,op.name,c.firstname,c.lastname FROM `" . DB_PREFIX . "razorpay_subscriptions` s";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "razorpay_plans p ON (p.entity_id = s.plan_entity_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "product_description op ON (op.product_id = p.opencart_product_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "customer c ON (s.opencart_user_id = c.customer_id )";
+        
+        $sql .= " WHERE s.entity_id= '" .$entity_id . "'";
+        $query = $this->db->query($sql);
+
+		return $query->row;
+	}
+    public function resumeSubscription($entity_id,$updated_by)
+    {
+        $this->db->query("UPDATE " . DB_PREFIX . "razorpay_subscriptions SET status = 'active',updated_by = '".$updated_by. "' WHERE entity_id = '" .$entity_id . "'");
+    }
+    public function pauseSubscription($entity_id)
+    {
+        $this->db->query("UPDATE " . DB_PREFIX . "razorpay_subscriptions SET status = 'paused',updated_by = '".$updated_by. "' WHERE entity_id = '" .$entity_id . "'");
+    }
+    public function cancelSubscription($entity_id)
+    {
+        $this->db->query("UPDATE " . DB_PREFIX . "razorpay_subscriptions SET status = 'cancelled' ,updated_by = '".$updated_by. "' WHERE entity_id = '" .$entity_id . "'");
+    }
```

Comment:
Merge these 3 functions into 1 function and pass the required parameters from controller from where your calling the functions